docs/proposals/improve-addons.md

```diff
@@ -0,0 +1,350 @@
+# Improve Addons
+
+**Author**: @vgramer @SimonTheLeg
+
+**Status**: Proposal
+
+## Table of Contents
+
+- [Improve Addons](#improve-addons)
+  - [Table of Contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Motivation and Background](#motivation-and-background)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Implementation](#implementation)
+  - [Architecture](#architecture)
+  - [Components](#components)
+    - [AddonCatalogueDistributor](#addoncataloguedistributor)
+    - [AddonCatalogue CR](#addoncatalogue-cr)
+    - [AddonController](#addoncontroller)
+    - [AddonInstallation CR](#addoninstallation-cr)
+    - [CoreAddonController](#coreaddoncontroller)
+    - [AddonInstallationController](#addoninstallationcontroller)
+  - [Common User Flows](#common-user-flows)
+  - [Alternatives considered](#alternatives-considered)
+- [Glossary](#glossary)
+  - [CoreAddon (old name: Default **Addon**)](#coreaddon-old-name-default-addon)
+  - [OptionalAddon (old name: Accessible Addon)](#optionaladdon-old-name-accessible-addon)
+  - [CustomerAddon (old name: Custom Addon)](#customeraddon-old-name-custom-addon)
+- [Task & effort](#task--effort)
+
+## Introduction
+
+This proposal introduces improvements to the current [Addon system](https://docs.kubermatic.com/kubermatic/v2.18/guides/addons/). It aims to make current addons more flexible and user-friendly. Additionally, it simplifies the usage of addons we ship with (almost) every cluster, so-called "Default Addons".
+
+## Motivation and Background
+
+Currently, KKP only supports one mechanism for addons: For this, all addons are stored in a single docker image that is configured in KKP. Afterwards, an initContainer is being run that starts the image and copies all addons onto the seed-controller-managers local file-system. From there, addon manifests are being rendered and rolled out to user clusters.
+
+The current implementation of Addons has some flaws that make the usage and maintenance of Addons cumbersome for cluster administrators:
+
+- To add “Custom Addons” the cluster-admin needs to bundle manifests and templates for these Addons in a Docker image that is based on the official addon-manifest docker image by Kubermatic.
```

Comment:
Besides that in most of the cases user needs to create AddonConfig CRD for each custom/accessible addon, so form, logo and description will be visible in the dashboard.

Reply:
See: https://github.com/kubermatic/kubermatic/issues/5499